src/optimize-streams.ts

```diff
@@ -113,28 +114,37 @@ export class GenericOptimizeTransform extends Transform {
  */
 export class JsTransform extends GenericOptimizeTransform {
   constructor(options: OptimizeOptions['js']) {
+    const shouldCompileFile =
+        options.compile ? notExcluded(options.compile) : () => false;
+    const shouldMinifyFile =
+        options.minify ? notExcluded(options.minify) : () => false;
+    const nodeResolution = options.moduleResolution === 'node';
+
     const transformer = (content: string, file: File) => {
+      // Note that Babel plugins run in this order:
+      // 1) plugins, first to last
+      // 2) presets, last to first
+      const plugins = [...babelSyntaxPlugins];
       const presets = [];
-      const plugins = [
-        // Syntax plugins for >ES2015 features we support.
-        objectRestSpreadSyntax,
-        dynamicImportSyntax,
-      ];
-      if (options.compile) {
+
+      if (shouldMinifyFile(file)) {
+        // Minify last, so push first.
+        presets.push(babelPresetMinify);
+      }
+      if (shouldCompileFile(file)) {
         presets.push(babelPresetES2015NoModules);
         plugins.push(
-            externalHelpersPlugin,
-            objectRestSpreadTransform,
+            babelPluginExternalHelpers,
+            babelTransformPluginObjectRestSpread,
         );
       }
-      if (options.moduleResolution === 'node') {
+      if (nodeResolution) {
```

Comment:
why not  `if (options.moduleResolution === 'node') {`, or ` if (options.moduleResolution !== 'node') {`? At least do an explicit check against undefined.

Reply:
Done.